### DISCUSSION TITLE
Dynamic locale import with webpackMode=lazy results in hundreds of chunks

### DISCUSSION THREAD

**Discussion by jbaldassari** [js-fns/date-fns#3284] [General]
I just wanted to post this here in case it helps anyone else.  I noticed that when dynamically importing locales with webpack and using the default options (no mode specified or `webpackMode: "lazy"`), webpack generates 500+ chunks which are loaded serially, significantly impacting app load time.

Example:

```
await import(`date-fns/locale/${locale}/index.js`);
```

To avoid this, pass the webpack hint `webpackMode: "lazy-once"` (see docs [here](https://webpack.js.org/api/module-methods/#magic-comments)), which still keeps the locales out of the main chunk but causes all of the locales to be loaded together as a single chunk instead of individual chunks.  This results in much faster load time.  Example:

```
await import(
  /* webpackMode: "lazy-once", webpackChunkName: "date-fns" */ `date-fns/locale/${resolvedLocaleCode}/index.js`
);
```

The before and after screenshots from `source-map-explorer` are below:

**Before:**

![image](https://user-images.githubusercontent.com/307312/204117172-3172a30b-4aa9-4d6a-aa2e-1af368725f62.png)

**After:**

![image](https://user-images.githubusercontent.com/307312/204117188-8d83718a-3915-41e2-9bd2-cbf771cb1326.png)

**Comment by joaopedrodcf** [comment]
Having them separated and loaded async isn't better ?
